opendevin/controller/agent_controller.py

```diff
@@ -108,7 +109,7 @@ async def _start_step_loop(self):
             except Exception as e:
                 logger.error(f'Error while running the agent: {e}')
                 await self.report_error(
-                    'There was an unexpected error while running the agent'
+                    'There was an unexpected error while running the agent: {e}'
```

Comment:
Just a small question, is this string going to get to the UI? If yes, we stopped sending the actual exception there, it's only logged. But maybe it doesn't go there, sorry I don't see clearly atm

Reply:
Oh yeah good catch, this is indeed sent to UI. Aside, there's ` await self.report_error(str(e))` in this file which also sends the actual exception to the UI.

Is there any reason for not sending the actual exception? Sorry I might have missed previous discussions.

Reply:
If we don't want to send it to the UI, maybe we could make exception `e` as an separate argument and only set that to state w/o affecting UI?

Reply:
> there's  await self.report_error(str(e)) in this file which also sends the actual exception to the UI.

You're right, but please note that the occurrence of `self.report_error(str(e))` is on _our_ converted exceptions only, which have a user-friendly string. Any of those exceptions caught in that place is actually a conversion from raw exception to something with a cleaned-up message for the end-user. Informative, but not too raw. The logs, not the UI, are for "code-level" raw information. Most users are not developers, the UI message is for their reading.

If I may put the idea here in my silly words, it's like ... maybe the end-user interface should not be our debugging playground 😺 

It's a bit difficult to list all this is coming from. The actual change was here: https://github.com/OpenDevin/OpenDevin/pull/1296

Reply:
@xingyaoww SGTM? I understand that if we don't send `str{e}` here, at this line, the eval is missing information it needs?

I've been playing with the idea of introducing a "cause" attribute to our custom exceptions, so that we can split this, send over the underlying exception to reach the LLM and the `message` to reach the UI... I've put it in the back burner, and meanwhile we got a `cause` attribute with a different meaning, but if it helps, maybe we can ressurrect that in some form

Reply:
How about now :)

Reply:
I can't test right now, but if y'all are happy with it, I'm happy with it!